packages/schema/src/config/build.ts

```diff
@@ -38,6 +39,18 @@ export default defineUntypedSchema({
     },
   },
 
+  /**
+   * Log level when building logs.
+   *
+   * Defaults to 'silent' when running in CI or when a TTY is not available.
+   * This option is then used as 'silent' in Vite and 'none' in Webpack
+   *
+   * @type {'silent' | 'info' | 'verbose'}
+   */
+  logLevel: {
+    $resolve: (val) => val ?? (isTest ? 'silent' : 'info')
```

Comment:
We probably need to validate val being in allowed list [silent,info, verbose] otherwise mapping logic in other places can be unexpected

Reply:
I think with default value provided at location of map, we can probably omit the validation here and leave it to the schema to type-check for us.

Reply:
It does for config i agree. But CLI is passing user args as is `--logLevel foobar` won't be validated. And it can go unnoticed.

Reply:
I think both worth to be in place and making sure nuxt.options is predictable. A new log level can be introduced and easily forgotten to be added to the maps. Furthor in any external module that consumes this new option.

Reply:
The way I've typed the maps, if a new log level is added without a map entry, it will cause a nuxt build failure.

Reply:
Types are not enough as external modules can depend on different schema. We need validation here. What is the downside of doing it?

Reply:
My point is not that we shouldn't validate, but that if we do, we do not need to provide a default value for the maps as we _know_ that the value validated by the schema exists in the map (because it will be a TS if it does not).

Reply:
I get your point. First, for schema validation, we need it because:

- A user can pass a wrongly/unsupported type via `nuxi dev --logLevel xyz`
- We keep it invalidated in `nuxt.options.logLevel` as `xyz`
- Nuxt internals silently handle a fallback
- Nuxt modules (3rd party) depending on `logLevel` can do unexpected behavior, trusting the type is within 3 possible values

Second, as **best practice** (as mentioned in other comment), we use a fallback. If an adhoc-module let's say supports 3 current types, and Nuxt version by user comes with a new supported type `warn`, the module fails. Again, i understand this won't happen considering we have both runtime and static type validation in place but it gives implementation consistency.

Feel free to revert first commit if you feel it would be extra even with this explanation :) Schema validation is probably most important part.

Reply:
I'm not disagreeing on validation, though I would note that your point about modules depending on the schema is really rather broader than this PR. The rest of the schema is not validated for the most part and all the comments you've just made about this _particular_ option are also relevant there too. Maybe worth adding some schema validation primitives upstream to `c12`?

Reply:
It is special because is one of the first options coming directly from CLI args and they cannot be type-checked. And we do have normalization for some of other configs.

We definitely have to practice validation checks better (i guess `untyped` for auto validation).